bip-0173.mediawiki

```diff
@@ -181,12 +181,15 @@ in the string an error might be found, without suggesting the correction
 to make.
 
 '''Uppercase/lowercase'''
+The lowercase form is used when determining a character's value for checksum purposes.
 
-Decoders MUST accept both uppercase and lowercase strings, but
-not mixed case. The lowercase form is used when determining a character's
-value for checksum purposes. For presentation, lowercase is usually
-preferable, but inside QR codes uppercase SHOULD be used, as those permit
-the use of
+Encoders MUST always output an all lowercase Bech32 string.
+If an uppercase version of the encoding result is desired, (e.g.- for presentation purposes, or QR code use),
+then an uppercasing procedure can be performed external to the encoding process.
+
+Decoders MUST accept uppercase, lowercase, and mixed-case strings.
```

Comment:
Decoders must not accept mixed-case strings...

Reply:
Not a typo. I think decoders should (and thus "must") accept mixed-case. Because of the existing requirement to use the lowercase form to determine a character's value for checksum purposes, there is no difference between upper and lower case characters in any position of the bech32 string. That applies to the HRP because it is also used to compute the checksum. So, on a micro level, I don't see a reason why a decoder should fail the input. On a macro level, I can see where you might be trying to discourage mixed case. But, the other changes I made also push in that direction (in fact fixing this: `segwit_addr.bech32_encode('A', [])`).